rust/otap-dataflow/crates/engine/src/pipeline_metrics.rs

```diff
@@ -18,7 +18,7 @@
 //!   Wall-clock time since this pipeline instance started. Useful for detecting
 //!   restarts, warm-up effects, and aligning deltas to a known lifetime.
 //!
-//! - `cpu_time` (`Counter<f64>`, `{s}`):
+//! - `cpu_time` (`DeltaCounter<f64>`, `{s}`):
```

Comment:
For the record (not to block or change) I think you could keep `Counter` as meaning delta counter and use `ObserveCounter` as the cumulative form, likewise for UpDown forms. I think we can have this added in the OTel specification at some point in the future, and these names reflect my interpretation of the OTel convention. Whereas you Add() (and Sub()) a Counter (and UpDownCounter) you always Set() an Observe(UpDown)Counter, like the synchronous Gauge.